### PR TITLE
libxmlsec1 1.3.3 (include patch for lasso)

### PR DESCRIPTION
**Pull request by timsutton**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Follow-up to https://github.com/Homebrew/homebrew-core/pull/159024

### CODE REVIEW

File: Formula/l/lasso.rb [hunk 1]
```diff
@@ -35,17 +35,23 @@ class Lasso < Formula
     depends_on "six" => :build # macOS Python has `six` installed.
   end
 
+  # patch from upstream issue: https://dev.entrouvert.org/issues/85339
+  # Remove when https://git.entrouvert.org/entrouvert/lasso/pulls/10/ is merged
+  patch :DATA
+
   def install
+    # Fix compile with newer Clang
+    ENV.append_to_cflags "-Wno-implicit-function-declaration" if DevelopmentTools.clang_build_version >= 1403
```
Comment (Bo98) [inline Formula/l/lasso.rb:44]: This is suspicious given it worked before. This suggests the patch has broken this and is probably a bug in the patch.

Comment (timsutton) [reply Formula/l/lasso.rb:44]: This actually wasn't related to the patch and I didn't seem to need it locally. I had noticed it in @chenrui333's earlier fixup commit [here](https://github.com/Homebrew/homebrew-core/pull/159024/commits/966b5f8a9520f8ef8ea18217ae6633bb3944474b) and decided to just let it run through CI first.

Comment (Bo98) [reply Formula/l/lasso.rb:44]: In that case, the error was likely signalling a genuine compatibility issue and was telling you about the functions removed in 1.3.3, which the patch fixes. The flag shouldn't be necessary if so.

Comment (timsutton) [reply Formula/l/lasso.rb]: Agreed – 1.3.3 definitely [removed those functions](https://github.com/lsh123/xmlsec/compare/xmlsec_1_3_2...1.3.3#diff-358f54e061b73691c01260fd203beff99182c3743f5ec1d86b9ffb670814adccL718), they were already deprecated.

lasso's version was last bumped quite a while ago and so I wonder if maybe it just hadn't had to be built yet with a new enough clang/Xcode?

Comment (Bo98) [reply Formula/l/lasso.rb]: It will have for the Sonoma bottle.

**Review by Bo98 — APPROVED** [review]

**Review by chenrui333 — COMMENTED** [review]
👍

**Pull request merged**